lib/Installation/License/Sle/LicenseAgreementController.pm

```diff
@@ -15,16 +15,17 @@ use warnings;
 use Installation::License::AcceptLicensePopup;
 use Installation::License::LicenseAgreementExplicitPage;
 use YuiRestClient;
+use Utils::Architectures 'is_s390x';
```

Comment:
you need to implement other page/controller (perhaps inheriting from current one or not). There are not conditions `if/else` when designing with libyui-rest-api in general.

Reply:
I understand your point, but somewhere the decision which IDs are used for the widgets need to be used depending on is_s390x(). Where shall we place this? The test module invokes the Controller, so if we build up different Controllers of s390x and non-s390x then we need to put the decision in the test module and adapt more than when we decide to do the switch inside the controller.

Reply:
actually we should adapt less, I mean change existing code, but we might need more new files to do it, that is the benefit of having such a design. If we introduce conditions in the framework then the framework would start to look more like test using testapi and really hard to maintain. Let's have discussion offline about how to design it.

Reply:
I'm thinking if we can use this kind of design:
btn_ok_filter => {id => 'ok_msg|ok'} then we can find each 'id' in the find_widget, once we find one then we can return. if both can't be found then we can error out. this will make it arch-independent.

Reply:
actually that is a good option :) 
you can use regex for those filters, `id => qr/ok_msg|ok/`
it is not perfect solution, because those are different screens and with don't follow the POM strictly...but if you don't need to change anything else, is the case?